src/org/anjocaido/groupmanager/permissions/BukkitPermissions.java

```diff
@@ -350,10 +351,8 @@ public void updatePermissions(Player player, String world) {
 			}
 		}
 		alreadyVisited.remove(node);
-		if (!alreadyVisited.isEmpty())
-			return alreadyVisited;
 
-		return null;
+		return !alreadyVisited.isEmpty() ? alreadyVisited : null;

```

Comment:
you could remove the ! and reverse this logic to make it easier to read

`return alreadyVisited.isEmpty() ? null : alreadyVisited ;`

Reply:
will do